plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt

```diff
@@ -190,8 +195,30 @@ internal fun mapSnippetFindings(rawResults: RawResults, issues: MutableList<Issu
                 additionalSnippetData
             )
 
-            sourceLocations.forEach {
-                findings.getOrPut(it) { mutableSetOf(ortSnippet) } += ortSnippet
+            sourceLocations.forEach { sourceLocation ->
+                val isSnippetChoice = when {
+                    snippetChoices.any { it.given.sourceLocation == sourceLocation && it.choice.purl == purl } -> {
+                        logger.info {
+                            "Ignoring snippet $purl for file ${sourceLocation.prettyPrint()}, " +
+                                "as this is a chosen snippet."
+                        }
+                        true
+                    }
+
+                    snippetChoices.any { it.given.sourceLocation == sourceLocation } -> {
+                        logger.info {
+                            "Ignoring snippet $purl for file ${sourceLocation.prettyPrint()}, " +
+                                "as there is a snippet choice for this source location."
+                        }
+                        true
+                    }
+
+                    else -> false
+                }
+
+                if (!isSnippetChoice) {
+                    findings.getOrPut(sourceLocation) { mutableSetOf(ortSnippet) } += ortSnippet
+                }
```

Comment:
I tried to come up with something without the explicit `true`s and that does not iterate the whole list of choices twice, and ended up with this:

```kotlin
val choicesForLocation = snippetChoices.filter { it.given.sourceLocation == sourceLocation }
val choicesForLocationAndPurl = choicesForLocation.filter { it.choice.purl == purl }

val isChosenSnippet = (choicesForLocation.isNotEmpty() && choicesForLocationAndPurl.isNotEmpty()).also {
    logger.info {
        "Ignoring snippet $purl for file ${sourceLocation.prettyPrint()}, " +
            "as this is a chosen snippet."
    }
}

val isChosenSnippetLocation = choicesForLocation.isNotEmpty().also {
    logger.info {
        "Ignoring snippet $purl for file ${sourceLocation.prettyPrint()}, " +
            "as there is a snippet choice for this source location."
    }
}

if (!isChosenSnippet && !isChosenSnippetLocation) {
    findings.getOrPut(sourceLocation) { mutableSetOf(ortSnippet) } += ortSnippet
}
```

Not sure if it's much nicer, though.

Reply:
It creates double logging in case `choicesForLocation.isNotEmpty()` is `true`, while the previous code didn't.